diff --git a/index.js b/index.js
index a22558e5c0b2779ac875d14288f8d67c22889839..995d3e846a6da5899b12cf34b96d4bd12c664060 100644
--- a/index.js
+++ b/index.js
@@ -5,6 +5,7 @@ const { ValidationError } = require("@lerna/validation-error");
 const { CyclicPackageGraphNode } = require("./lib/cyclic-package-graph-node");
 const { PackageGraphNode } = require("./lib/package-graph-node");
 const { reportCycles } = require("./lib/report-cycles");
+const { Project } = require("@lerna/project");
 
 /** @typedef {import("./lib/package-graph-node").PackageGraphNode} PackageGraphNode */
 
@@ -22,8 +23,14 @@ class PackageGraph extends Map {
    * @param {boolean} [forceLocal] Force all local dependencies to be linked.
    */
   constructor(packages, graphType = "allDependencies", forceLocal) {
+  
+  
     super(packages.map((pkg) => [pkg.name, new PackageGraphNode(pkg)]));
 
+    const project = new Project()
+    const allPackages = project.getPackagesSync()
+    const allPackagesMap = new Map(allPackages.map((pkg) => [pkg.name, new PackageGraphNode(pkg)]));
+
     if (packages.length !== this.size) {
       // weed out the duplicates
       const seen = new Map();
@@ -58,7 +65,7 @@ class PackageGraph extends Map {
             );
 
       Object.keys(graphDependencies).forEach((depName) => {
-        const depNode = this.get(depName);
+        const depNode = allPackagesMap.get(depName);
         // Yarn decided to ignore https://github.com/npm/npm/pull/15900 and implemented "link:"
         // As they apparently have no intention of being compatible, we have to do it for them.
         // @see https://github.com/yarnpkg/yarn/issues/4212
